ynput/OpenPype
# Add `rez` packages generation for `server_addons`

This commit will add a `python` script that allows us to quickly generate packages for `ayon-backend` ingestion, either by copying them to the `rez` repo (not advised) or installing them via the `ayon-frontend` by uploading the resulting `zip`s of the script.

## Changelog Description
This is a PR that will allow us to then test the functionality of this `ayon-backend` PR: https://github.com/ynput/ayon-backend/pull/29

Instead of creating a folder structure at this step, we delegate it to the `rezbuild.py` on the `ayon-backend` which expects the `package.py` and the usual folder structure  of the addon source code (https://github.com/ynput/ayon-addon-template), thus this step is greatly simplified where we only copy the contents of the addon, add a `package.py` and zip them up. The caveat of the `openpype` package is the same, so an extra step to copy the required things of OpenPype into the `client` folder.

## Testing notes:
1. Clone this branch
2. Run `python create_ayon_rez_addons.py`
3.  Confirm `zip` archives have been generated in `OpenPype/server_addon/rez_packages`
4. If you are running the backend with the rez implementation: Install the addon, enable it and make sure it is able to initialize.

##  TODO

- [ ] Test all individual addons and change where required so they initialize in the backend
